Decode/PerfItemType.cs

```diff
@@ -103,37 +164,52 @@ namespace Microsoft.LinuxTracepoints.Decode
         /// <summary>
         /// For simple encodings (e.g. Value8, Value16, Value32, Value64, Value128),
         /// this is the size of one element in bytes (1, 2, 4, 8, 16). For complex types
-        /// (e.g. Struct or string), this is 0.
+        /// (e.g. Struct or String), this is 0.
         /// </summary>
         public byte TypeSize { get; }
 
+        /// <summary>
+        /// Returns Encoding | ArrayFlag | (IsScalar ? ChainFlag : 0).
+        /// </summary>
+        private EventHeaderFieldEncoding EncodingAndArrayFlagAndIsScalar { get; }
+
         /// <summary>
         /// Item's underlying encoding. The encoding indicates how to determine the item's
         /// size. The Encoding also implies a default formatting that should be used if
-        /// the specified convertOptions is Default (0), unrecognized, or unsupported. The value
+        /// the specified Format is Default (0), unrecognized, or unsupported. The value
         /// returned by this property does not include any flags.
         /// </summary>
         public EventHeaderFieldEncoding Encoding =>
-            this.EncodingAndArrayFlags & EventHeaderFieldEncoding.ValueMask;
+            this.EncodingAndArrayFlagAndIsScalar & EventHeaderFieldEncoding.ValueMask;
 
         /// <summary>
-        /// Contains CArrayFlag or VArrayFlag if the item represents an array begin,
-        /// array end, or an element within an array. 0 for a non-array item.
+        /// Returns the field's CArrayFlag or VArrayFlag if the item represents an array-begin
+        /// field, an array-end field, or an element within an array field.
+        /// Returns 0 for a non-array item.
         /// </summary>
-        public EventHeaderFieldEncoding ArrayFlags =>
-            this.EncodingAndArrayFlags & ~EventHeaderFieldEncoding.ValueMask;
+        public EventHeaderFieldEncoding ArrayFlag =>
+            this.EncodingAndArrayFlagAndIsScalar & EventHeaderFieldEncoding.ArrayFlagMask;
```

Comment:
Note: Since only one array flag can be set at a time, rename the `ArrayFlags` property to `ArrayFlag`.